### PR TITLE
WIP : Initial test for gitlab integration

### PR DESCRIPTION
**Pull request by xdecock**
Well basicaly a preparation of our plugin to run with gitlab instead of jira.

I'm not certain about how to change the enabled plugins, so mostly there to get initial feedback.

And thx for the work.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -410,7 +410,8 @@ def run(self):
             "google_drive_task = dispatch.plugins.dispatch_google.drive.plugin:GoogleDriveTaskPlugin",
             "google_gmail_conversation = dispatch.plugins.dispatch_google.gmail.plugin:GoogleGmailConversationPlugin",
             "google_groups_participants = dispatch.plugins.dispatch_google.groups.plugin:GoogleGroupParticipantGroupPlugin",
-            "jira_ticket = dispatch.plugins.dispatch_jira.plugin:JiraTicketPlugin",
+            #"jira_ticket = dispatch.plugins.dispatch_jira.plugin:JiraTicketPlugin",
```
Comment (mvilanova) [inline setup.py]: No need to disable the Jira plugin. You can tell Dispatch to use your Gitlab plugin by adding `INCIDENT_PLUGIN_TICKET_SLUG="gitlab-ticket"` to your `.env` file. Dispatch uses the Jira one by default if none are specified. See https://github.com/Netflix/dispatch/blob/develop/src/dispatch/config.py#L90

File: src/dispatch/plugins/dispatch_gitlab/config.py [hunk 2]
```diff
@@ -0,0 +1,9 @@
+from starlette.datastructures import URL
+
+from dispatch.config import config, Secret
+
+
+GITLAB_BROWSER_URL = config("GITLAB_BROWSER_URL", cast=URL)
+GITLAB_API_URL = config("GITLAB_API_URL", cast=URL)
+GITLAB_AUTH_KEY = config("GITLAB_AUTH_KEY")
```
Comment (mvilanova) [inline src/dispatch/plugins/dispatch_gitlab/config.py]: Add `cast=Secret`, if it's a secret.

File: src/dispatch/plugins/dispatch_gitlab/plugin.py [hunk 3]
```diff
@@ -0,0 +1,193 @@
+"""
+.. module: dispatch.plugins.dispatch_jira.plugin
```
Comment (mvilanova) [inline src/dispatch/plugins/dispatch_gitlab/plugin.py]: Replace `dispatch_jira` with dispatch_gitlab`.

**Comment by mvilanova** [comment]
@xdecock thanks for your contribution. Would you mind adding documentation for this plugin? Examples here: https://github.com/Netflix/dispatch/tree/develop/docs/configuration/plugins

**Comment by xdecock** [comment]
Wow thanks for the feedback.

I'll apply the suggestions, try to run the plugin on our side along with the full dispatch I'll need to mock the pager plugin first. As we're running victor ops. So that's probably the next plugin we'll provide.

And do the documentation bit to get this going. Further.

File: setup.py (hunk 1 above)
Comment (xdecock) [reply setup.py]: Fixed, thx for the infos.

File: src/dispatch/plugins/dispatch_gitlab/config.py (hunk 2 above)
Comment (xdecock) [reply src/dispatch/plugins/dispatch_gitlab/config.py]: :heavy_check_mark:

**Comment by kevgliss** [comment]
Looks good to me. Will be awesome to have support for Gitlab!

**Comment by mvilanova** [comment]
@xdecock we're almost there. We will merge the code once you fix the Black and Flake8 errors.

**Comment by xdecock** [comment]
So fixed, styling errors and warnings.

Code is still largely living in that marvelous country called theory, So basically not tested live, will probably get time next week to start deploying the system and get real datas through it, yet, if somebody with more courage and time than me wants to get the first bugs out, i'll gladly help to fix them, wasn't expecting the pr to go that fast.

**Comment by kevgliss** [comment]
@xdecock Closing this for now, feel free to re-open when you get some spare cycles to test it out.

**Pull request closed without merging**